lib/strong_resources/controller/mixin.rb

```diff
@@ -21,10 +22,24 @@ def update_action?
         action_name == 'update'
       end
 
+      private
+
+      def _params_type
+        params_method = if respond_to?(:raw_params) 
+                          :raw_params 
+                        else
+                          :params
+                        end
+
+        send(params_method).try(:[], :data).try(:[], :type).try(:to_sym)
+      end
+
       module ClassMethods
         def strong_resource(name, opts = {}, &blk)
+          blk ||= Proc.new {}
           opts[:require] ||= name unless opts[:require] == false
           resource = StrongResource.from(name, opts, &blk)
+          type = opts[:type] || name
```

Comment:
the `type` in jsonapi payload is usually the pluralized name. But my main point is that logic already exists elswhere in the gem, and this would be duplicating the logic. We need "inferring type" all in one place.